### PR TITLE
{RDBMS} Fixed bugs and updated local context test

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

- Bug fix: not invoking error when command fails. Currently only log error message which will not give 'return 1' error when running a script with azure cli commands. 
- Bug fix: az mysql/postgres flexible-server delete command don't output error when server name is not fed
- Test update: local context test update to properly validate the feature

**Testing Guide**

azdev test FlexibleServerLocalContextScenarioTest

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by DaeunYim** [comment]
CI pass fixed. Requesting review for this PR

**Comment by yonzhan** [comment]
@evelyn-ys will help you with this.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 1]
```diff
@@ -41,98 +41,88 @@ def flexible_server_create(cmd, client, resource_group_name=None, server_name=No
...
-                               "--vnet, --subnet, --vnet-address-prefix, --subnet-address-prefix must be supplied together.")
-
-        server_result = firewall_id = subnet_id = None
-
-        # Check availability for server name if it is supplied by the user
-        if server_name is not None:
-            check_name_client = cf_mysql_check_resource_availability(cmd.cli_ctx, None)
-            server_availability = check_name_client.execute(server_name, DELEGATION_SERVICE_NAME)
-            if not server_availability.name_available:
-                raise CLIError(server_availability.message)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:71]: Why delete the check for server name?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 2]
```diff
@@ -256,8 +246,14 @@ def flexible_server_update_custom_func(cmd, instance,
 
 
 def server_delete_func(cmd, client, resource_group_name=None, server_name=None, yes=None):
+    if resource_group_name is None:
+        raise CLIError('Please specify the name of resource group.')
+    if server_name is None:
+        raise CLIError('Please specify the name of server you want to delete.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:252]:
```suggestion
def server_delete_func(cmd, client, resource_group_name, server_name, yes=None):
```
Do you want to make `resource_group_name` and `server_name` required here?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py [hunk 3]
```diff
@@ -46,91 +46,82 @@ def flexible_server_create(cmd, client,
...
-                               "--vnet, --subnet, --vnet-address-prefix, --subnet-address-prefix must be supplied together.")
-
-        server_result = firewall_id = subnet_id = None
-
-        # Check availability for server name if it is supplied by the user
-        if server_name is not None:
-            check_name_client = cf_postgres_check_resource_availability(cmd.cli_ctx, None)
-            server_availability = check_name_client.execute(server_name, DELEGATION_SERVICE_NAME)
-            if not server_availability.name_available:
-                raise CLIError(server_availability.message)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py:76]: Same as above

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:71]: It's duplicated and causing error. The name check will be done inside _create_server function.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py (hunk 3 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py:76]: Same as above

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py (hunk 2 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:252]: You're right. I'll make the change per your comment. Thanks!

**Comment by DaeunYim** [comment]
@evelyn-ys Can this be merged for this train? :)

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**